book/maps-and-hashtables/README.md

```diff
@@ -642,6 +642,68 @@ you can always write your own comparison function by hand; but if all you
 need is a total order suitable for creating maps and sets with, then
 `[@@deriving compare]` is a good choice.
 
+If you use Core rather than Base, things are a bit different. Either `Comparator.Make` 
+or `Comparable.Make` requires `t_of_sexp` for the generated serialization modules. 
+We need to change the deriving from `sexp_of` to `sexp`.
+
+```ocaml env=core_compare
+# #require "ppx_jane"
+# open Core
+# module Book = struct
+    module T = struct
+      type t = { title: string; isbn: string }
+      [@@deriving compare, sexp]
+    end
+    include T
+    include Comparator.Make(T)
+  end
+module Book :
+  sig
+    module T :
+      sig
+        type t = { title : string; isbn : string; }
+        val compare : t -> t -> int
+        val t_of_sexp : Sexplib0.Sexp.t -> t
+        val sexp_of_t : t -> Sexp.t
+      end
+    type t = T.t = { title : string; isbn : string; }
+    val compare : t -> t -> int
+    val t_of_sexp : Sexp.t -> t
+    val sexp_of_t : t -> Sexp.t
+    type comparator_witness = Base.Comparator.Make(T).comparator_witness
+    val comparator : (t, comparator_witness) Comparator.t
+  end
+```
+
+`Core.Comparable` generates inner modules `Map` and `Set` for serialization. When you open this module locally, the `Core.Set` will be shadowed by 
```

Comment:
Hmm.  I'm not excited about this way of talking about it, because the real mistake in the code here is the local open of Book. Generally speaking, opening modules should be minimized, and there's really no justification for opening Book here.

I agree that it would be good to have a little subsection explaining the differences between Base and Core/Core_kernel, but I don't think this is quite getting at it.  I think I'll take a separate swing at this in a new PR.

Reply:
It's subtle here but I agree that the code is not expressive enough. I read the [open modules section](https://dev.realworldocaml.org/files-modules-and-programs.html#opening-modules) and the intended scenario is like `print_median`, e.g. `collect_median` where you add the median to another Set or Map. I will back on this when I have a good example for this.

I will make a separate PR, just explaining the differences in short words later.